diff --git a/src/libsodium/crypto_stream/chacha20/ref/chacha20_ref.c b/src/libsodium/crypto_stream/chacha20/ref/chacha20_ref.c
index 40cccbf8..fd2f120a 100644
--- a/src/libsodium/crypto_stream/chacha20/ref/chacha20_ref.c
+++ b/src/libsodium/crypto_stream/chacha20/ref/chacha20_ref.c
@@ -221,7 +221,7 @@ chacha20_encrypt_bytes(chacha_ctx *ctx, const uint8_t *m, uint8_t *c,
 }
 
 static int
-stream_ref(unsigned char *c, unsigned long long clen, const unsigned char *n,
+stream_ref_ref(unsigned char *c, unsigned long long clen, const unsigned char *n,
            const unsigned char *k)
 {
     struct chacha_ctx ctx;
@@ -259,7 +259,7 @@ stream_ietf_ext_ref(unsigned char *c, unsigned long long clen,
 }
 
 static int
-stream_ref_xor_ic(unsigned char *c, const unsigned char *m,
+stream_ref_xor_ic_ref(unsigned char *c, const unsigned char *m,
                   unsigned long long mlen, const unsigned char *n, uint64_t ic,
                   const unsigned char *k)
 {
@@ -305,8 +305,8 @@ stream_ietf_ext_ref_xor_ic(unsigned char *c, const unsigned char *m,
 
 struct crypto_stream_chacha20_implementation
     crypto_stream_chacha20_ref_implementation = {
-        SODIUM_C99(.stream =) stream_ref,
+        SODIUM_C99(.stream =) stream_ref_ref,
         SODIUM_C99(.stream_ietf_ext =) stream_ietf_ext_ref,
-        SODIUM_C99(.stream_xor_ic =) stream_ref_xor_ic,
+        SODIUM_C99(.stream_xor_ic =) stream_ref_xor_ic_ref,
         SODIUM_C99(.stream_ietf_ext_xor_ic =) stream_ietf_ext_ref_xor_ic
     };
